### PR TITLE
Dousi Server add close method.

### PR DESCRIPTION
**Pull request by meijies**

### CODE REVIEW

**Comment by jovany-wang** [comment]
# [Codecov](https://codecov.io/gh/jovany-wang/dousi/pull/111?src=pr&el=h1) Report
> Merging [#111](https://codecov.io/gh/jovany-wang/dousi/pull/111?src=pr&el=desc) into [master](https://codecov.io/gh/jovany-wang/dousi/commit/28a386f3c40f7ab06a38e8cac84193c14dfeef8d&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/jovany-wang/dousi/pull/111?src=pr&el=tree)

```diff
@@            Coverage Diff            @@
##             master     #111   +/-   ##
=========================================
  Coverage     16.92%   16.92%           
  Complexity        8        8           
=========================================
  Files             6        6           
  Lines           130      130           
  Branches          6        6           
=========================================
  Hits             22       22           
  Misses          107      107           
  Partials          1        1           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/jovany-wang/dousi/pull/111?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/jovany-wang/dousi/pull/111?src=pr&el=footer). Last update [28a386f...5d1f1a8](https://codecov.io/gh/jovany-wang/dousi/pull/111?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: protocol/src/main/java/org/dousi/DousiServer.java [hunk 1]
```diff
@@ -29,14 +35,23 @@ public DousiServer(ServerConfig serverConfig) {
...
+   * @param serviceObject  The object that this service implementation.
    */
   public <T> void registerService(Class<T> interfaceClass, T serviceObject) {
     serviceHandlers.add(new HandlerDelegate(new ServerImpl<T>(serviceObject, interfaceClass)));
   }
 
   public void run() {
-    Server server = NettyTransportFactory.getInstance().createServer(serverConfig, serviceHandlers);
+    if (server != null) {
+      logger.info("server is running");
```
Comment (jovany-wang) [inline protocol/src/main/java/org/dousi/DousiServer.java]: info -> debug

File: protocol/src/main/java/org/dousi/DousiServer.java [hunk 2]
```diff
@@ -29,14 +35,23 @@ public DousiServer(ServerConfig serverConfig) {
    * Register the service into this exporter.
    *
    * @param interfaceClass The interface that we want to export.
-   * @param serviceObject The object that this service implementation.
+   * @param serviceObject  The object that this service implementation.
```
Comment (jovany-wang) [inline protocol/src/main/java/org/dousi/DousiServer.java]: Why change this line?

Comment (meijies) [reply protocol/src/main/java/org/dousi/DousiServer.java]: idea format. I have reverted this change.

File: protocol/src/main/java/org/dousi/DousiServer.java (hunk 1 above)
Comment (meijies) [reply protocol/src/main/java/org/dousi/DousiServer.java]: fixed.

**Review by jovany-wang — APPROVED** [review]
LGTM

**Pull request merged**